src/ImageSharp/Formats/Webp/Lossy/YuvConversion.cs

```diff
@@ -312,6 +602,178 @@ public static void YuvToBgr(int y, int u, int v, Span<byte> bgr)
+                out Vector128<byte> g1,
+                out Vector128<byte> g2,
+                out Vector128<byte> g3,
+                out Vector128<byte> g4,
+                out Vector128<byte> g5);
+
+            // Process B.
+            ChannelMixing(
+                input4,
+                input5,
+                PlanarTo24Shuffle6,
+                PlanarTo24Shuffle7,
+                PlanarTo24Shuffle8,
+                out Vector128<byte> b0,
+                out Vector128<byte> b1,
+                out Vector128<byte> b2,
+                out Vector128<byte> b3,
+                out Vector128<byte> b4,
+                out Vector128<byte> b5);
+
+            // OR the different channels.
+            Vector128<byte> rg0 = Sse2.Or(r0, g0);
+            Vector128<byte> rg1 = Sse2.Or(r1, g1);
+            Vector128<byte> rg2 = Sse2.Or(r2, g2);
+            Vector128<byte> rg3 = Sse2.Or(r3, g3);
+            Vector128<byte> rg4 = Sse2.Or(r4, g4);
+            Vector128<byte> rg5 = Sse2.Or(r5, g5);
+
+            ref byte outputRef = ref MemoryMarshal.GetReference(rgb);
+            Unsafe.As<byte, Vector128<byte>>(ref outputRef) = Sse2.Or(rg0, b0);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 16)) = Sse2.Or(rg1, b1);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 32)) = Sse2.Or(rg2, b2);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 48)) = Sse2.Or(rg3, b3);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 64)) = Sse2.Or(rg4, b4);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 80)) = Sse2.Or(rg5, b5);
+        }
+
+        // Shuffles the input buffer as A0 0 0 A1 0 0 A2
+        private static void ChannelMixing(
+            Vector128<byte> input0,
+            Vector128<byte> input1,
+            Vector128<byte> shuffle0,
+            Vector128<byte> shuffle1,
+            Vector128<byte> shuffle2,
+            out Vector128<byte> output0,
+            out Vector128<byte> output1,
+            out Vector128<byte> output2,
+            out Vector128<byte> output3,
+            out Vector128<byte> output4,
+            out Vector128<byte> output5)
+        {
+            output0 = Ssse3.Shuffle(input0, shuffle0);
+            output1 = Ssse3.Shuffle(input0, shuffle1);
+            output2 = Ssse3.Shuffle(input0, shuffle2);
+            output3 = Ssse3.Shuffle(input1, shuffle0);
+            output4 = Ssse3.Shuffle(input1, shuffle1);
+            output5 = Ssse3.Shuffle(input1, shuffle2);
+        }
+
+        // Convert 32 samples of YUV444 to B/G/R
+        private static void ConvertYuv444ToBgrSse41(ref byte y, ref byte u, ref byte v, out Vector128<short> r, out Vector128<short> g, out Vector128<short> b)
+        {
+            Vector128<byte> y0 = LoadHigh(ref y);
+            Vector128<byte> u0 = LoadHigh(ref u);
+            Vector128<byte> v0 = LoadHigh(ref v);
```

Comment:
Wanted to stop already but can't resist 😄 Pipelined loads may perform better:

```suggestion
            Vector128<byte> y0 = Unsafe.As<byte, Vector128<byte>>(ref y);
            Vector128<byte> u0 = Unsafe.As<byte, Vector128<byte>>(ref u);
            Vector128<byte> v0 = Unsafe.As<byte, Vector128<byte>>(ref v);
            y0 = Sse2.UnpackLow(Vector128<byte>.Zero, y0);
            u0 = Sse2.UnpackLow(Vector128<byte>.Zero, u0);
            v0 = Sse2.UnpackLow(Vector128<byte>.Zero, v0);
```

Reply:
yes, makes sense. Changed with 7775c343